pkg/aws/actuator/actuator.go

```diff
@@ -840,7 +853,22 @@ func (a *AWSActuator) tagUser(logger log.FieldLogger, awsClient minteraws.Client
 		})
 	}
 
-	_, err := awsClient.TagUser(&iam.TagUserInput{
+	// appending the tags present in the Infrastructure CR
+	infra, err := utils.GetInfrastructure(a.Client)
+	if err != nil {
+		logger.WithError(err).Error("unable to get the infrastructure object")
```

Comment:
Why wouldn't the function just return an error here. Isn't the inability to read the Infrastructure resource a problem if this operator is supposed to be tagging things as specified in the Infrastructure?

Reply:
I was in a doubt here if this has to return an error or continue tagging the user with the other tags and log the error (as implemented above in an if/else statement).
Would make the required change. Thanks for pointing out.

Reply:
There is a bit of a potential problem if unable to retrieve the Infrastructure for multiple/repeated attempts, as the IAM User would never be tagged after creation, and then when someone goes to destroy the cluster, the IAM User will not be found because of the missing infrastructure tag. 
Perhaps way up in the syncMint() function where the LoadInfrastructureName() is called, you can also fetch the full Infrastructure resource and error before any AWS API calls are made https://github.com/openshift/cloud-credential-operator/blob/6ac05b473cdaf2b116a6432055dd4ed22aad6759/pkg/aws/actuator/actuator.go#L471

Reply:
Addressed and added the required code with a small refactoring :)